subsys/logging/log_core.c

```diff
@@ -173,25 +174,39 @@ static u32_t timestamp_get(void)
 	return k_cycle_get_32();
 }
 
-int log_init(void)
+void log_core_init(void)
+{
+	log_msg_pool_init();
+	log_list_init(&list);
+
+	/* No backends attached so far but set default level as a filter for
+	 * any source of logging in the system. When backends are attached later
+	 * logs will be filtered out during processing.
+	 */
+	if (IS_ENABLED(CONFIG_LOG_RUNTIME_FILTERING)) {
+		for (int i = 0; i < log_sources_count(); i++) {
+			u32_t *filters = log_dynamic_filters_get(i);
+
+			LOG_FILTER_SLOT_SET(filters,
+					    LOG_FILTER_AGGR_SLOT_IDX,
+					    CONFIG_LOG_DEFAULT_LEVEL);
+
+		}
+	}
+}
+
+void log_init(void)
 {
 	assert(log_backend_count_get() < LOG_FILTERS_NUM_OF_SLOTS);
 
 	/* Set default timestamp. */
 	timestamp_func = timestamp_get;
 	log_output_timestamp_freq_set(CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC);
 
-	if (!initialized) {
-		log_list_init(&list);
-
-		/* Assign ids to backends. */
-		for (int i = 0; i < log_backend_count_get(); i++) {
-			log_backend_id_set(log_backend_get(i),
-					 i + LOG_FILTER_FIRST_BACKEND_SLOT_IDX);
-		}
-
-		panic_mode = false;
-		initialized = true;
+	/* Assign ids to backends. */
+	for (int i = 0; i < log_backend_count_get(); i++) {
```

Comment:
Please introduce variables in the beginning of func

Reply:
done